lib/providers/generic.js

```diff
@@ -78,6 +82,13 @@ class GenericProvider {
   renew() {
     return this._retrieve();
   }
+
+  /**
+   * Removes cached data
+   */
+  invalidate() {
```

Comment:
Can we put the addition of an invalidate method for providers into its own pull request so it's not mixed in with the bug fix?